From 85dd77d6a6135e524a36bf31cdcb602ce41a05f0 Mon Sep 17 00:00:00 2001
From: Xingang Wang <wangxingang5@huawei.com>
Date: Thu, 23 Dec 2021 21:32:03 +0800
Subject: [PATCH] mpam: enable rdt_mon_capable for mbw monitor
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 85dd77d6a6135e524a36bf31cdcb602ce41a05f0


ascend inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4L735
CVE: NA

-------------------------------------------------

The rdt_mon_capable should be enabled when the msmon_mbwu
feature is supported, so that the mpam monitor function
can be effective.

Signed-off-by: Xingang Wang <wangxingang5@huawei.com>
Reviewed-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam/mpam_setup.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/mpam/mpam_setup.c b/arch/arm64/kernel/mpam/mpam_setup.c
index aae4a0be6304..d30910e0cda2 100644
--- a/arch/arm64/kernel/mpam/mpam_setup.c
+++ b/arch/arm64/kernel/mpam/mpam_setup.c
@@ -335,6 +335,7 @@ static void mpam_resctrl_pick_event_mbm_local(void)
 
 	if (mpam_has_feature(mpam_feat_msmon_mbwu, res->class->features)) {
 		res->resctrl_res.mon_capable = true;
+		rdt_mon_capable = true;
 		mpam_resctrl_events[QOS_L3_MBM_LOCAL_EVENT_ID] = *res;
 	}
 }
-- 
2.26.2

